vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/aggregator/FileAggregator.java

```diff
@@ -234,7 +235,7 @@ public ArtifactSet createArtifacts(Aggregate aggregate) throws RepositoryExcepti
             Artifact parent = new DirectoryArtifact(name, ".dir");
             artifacts.add(parent);
             // and extra
-            Serializer ser = new DocViewSerializer(aggregate);
+            Serializer ser = new DocViewSerializer((WalkableAggregate) aggregate);
```

Comment:
The issue, as I see it, is not solved, it simply gets shifted elsewhere - we still have a `cast` that probably could and definitely should be avoided :(

Reply:
yes. I agree. but the issue was about the cast to the `AggregateImpl` which made it difficult to use the serializer. if people want to use this library in a non-osgi environment, they can now implement their own `WalkableAggregates`, which wasn't possible before.